packages/amplify_api/example/ios/unit_tests/GraphQLApiUnitTests.swift

```diff
@@ -226,9 +228,151 @@ class GraphQLApiUnitTests: XCTestCase {
+                XCTFail()
+            }
+        }
+        
+        
+        FlutterGraphQLApi.subscribe(
+            flutterResult: { (result) -> Void in
+                if let exception = result as? FlutterError {
+                    XCTAssertEqual("AmplifyException", exception.code)
+                    XCTAssertEqual(FlutterApiErrorMessage.ESTABLISH_SUBSCRIPTION_FAILED.rawValue, exception.message)
+                    let errorMap: [String: Any] = exception.details as! [String : Any]
+                    XCTAssertEqual(
+                        ["platform": "iOS", "localizedErrorMessage": "test error", "recoverySuggestion": "test recovery suggestion"],
+                        errorMap["PLATFORM_EXCEPTIONS"] as? [String: String])
+                } else {
+                    XCTFail()
+                }
+            },
+            request: testRequest,
+            bridge: MockApiBridge(),
+            graphQLSubscriptionsStreamHandler: MockStreamHandler()
+        )
+    }
+    
+    func test_subscribe_malformed_request_error() throws {
+        let testRequest: [String: Any] = [:]
+        
+        class MockApiBridge: ApiBridge {
+            override func subscribe<R>(request: GraphQLRequest<R>,
+                                       valueListener: GraphQLSubscriptionOperation<R>.InProcessListener?,
+                                       completionListener: GraphQLSubscriptionOperation<R>.ResultListener?)
+                  -> GraphQLSubscriptionOperation<R> {
+                XCTFail("The ApiBridge should not be called if the request is malformed")
+                
+                let graphQLSubscriptionOperation = getMockGraphQLSubscriptionOperation(request: request)
+                return graphQLSubscriptionOperation
+            }
+        }
+        
+        class MockStreamHandler: GraphQLSubscriptionsStreamHandler {
+            override func sendEvent(payload: [String : Any?]?, id: String, type: GraphQLSubscriptionEventTypes) {
+                XCTFail()
+            }
+            
+            override func sendError(errorCode: String, details: [String: Any]) {
+                XCTFail()
+            }
+        }
+        
+        
+        FlutterGraphQLApi.subscribe(
+            flutterResult: { (result) -> Void in
+                if let exception = result as? FlutterError {
+                    XCTAssertEqual("AmplifyException", exception.code)
+                    XCTAssertEqual(FlutterApiErrorMessage.MALFORMED.rawValue, exception.message)
+                    let errorMap: [String: Any] = exception.details as! [String : Any]
+                    XCTAssertEqual(
+                        ["platform": "iOS", "localizedErrorMessage": "The graphQL document request argument was not passed as a String", "recoverySuggestion": "The request should include the graphQL document as a String"],
+                        errorMap["PLATFORM_EXCEPTIONS"] as? [String: String])
```

Comment:
Isn't this wrong?

Reply:
The tests were correct as per the original error handling approach. I've updated them now after the refactor.